src/formatDate.js

```diff
@@ -49,8 +49,43 @@
  * @returns {string}
  */
 
+const YY = 'YY';
+const YYYY = 'YYYY';
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const objDate = {};
+  const arrDateResult = [];
+  const delimiterFromFormat = fromFormat[3];
+  const delimiterToFormat = toFormat[3];
+  const arrDate = date.split(delimiterFromFormat);
+
+  arrDate.map((item, i) => {
+    objDate[fromFormat[i]] = item;
+  });
+
+  const keysDate = Object.keys(objDate);
+
+  toFormat.map((item, i) => {
```

Comment:
rather use forEach 

map is returning something, but seems not in that case


upd.: you can remove map method as well as returning true for 2 times 

and use only 2 if's conditions to check, if needed to swap YY and YYYY as well